### PR TITLE
add sign in logic

### PR DESCRIPTION
**Pull request by PogudoTanya**

### CODE REVIEW

File: src/resources/user/user.reducer.js [hunk 1]
```diff
@@ -13,6 +14,11 @@ export default (state = { userData: {}, pinCode: null }, action) => {
         ...state,
         authenticated: true,
       };
+    case USER_SIGNED_IN:
+      return {
+        ...state,
+        accessToken: action.userData.accessToken,
```
Comment (apsevkin) [inline src/resources/user/user.reducer.js:20]: let's set authenticated here to avoid 2 actions dispatch

File: src/screens/SignIn/SignIn.jsx [hunk 2]
```diff
@@ -30,13 +36,27 @@ function SignIn({ navigation }) {
     handleSubmit,
     setFieldValue,
     setFieldTouched,
+    setFieldError,
   } = useFormik({
-    onSubmit: () => {},
+    onSubmit: async () => {
+      try {
+        await dispatch(userActions.signIn(values));
```
Comment (apsevkin) [inline src/screens/SignIn/SignIn.jsx:43]: it would be better to avoid 2 dispatch calls

File: src/screens/SignIn/SignIn.jsx [hunk 3]
```diff
@@ -30,13 +36,27 @@ function SignIn({ navigation }) {
...
+        await dispatch(userActions.setUserAuthenticated());
+      } catch (e) {
+        if (e instanceof ApiError) {
+          if (e.data.credentials) {
+            setFieldError('password', e.data.credentials);
```
Comment (apsevkin) [inline src/screens/SignIn/SignIn.jsx:48]: let's add error only for one field, or add another error message after form fields

**Review by apsevkin — CHANGES_REQUESTED** [review]

File: src/screens/SignIn/SignIn.jsx [hunk 4]
```diff
@@ -30,13 +36,27 @@ function SignIn({ navigation }) {
...
+    setFieldError,
   } = useFormik({
-    onSubmit: () => {},
+    onSubmit: async () => {
+      try {
+        await dispatch(userActions.signIn(values));
+        await dispatch(userActions.setUserAuthenticated());
+      } catch (e) {
+        if (e instanceof ApiError) {
+          if (e.data.credentials) {
```
Comment (liubaterehova) [inline src/screens/SignIn/SignIn.jsx:47]: add e?.data.credentials

**Review by liubaterehova — APPROVED** [review]

**Review by liubaterehova — APPROVED** [review]

**Pull request merged**